src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -8,21 +8,33 @@
 
 from functools import wraps
 
-from mgr_module import MgrModule, HandleCommandResult, CLIWriteCommand, CLIReadCommand
+from mgr_module import MgrModule, HandleCommandResult, CLICommand
 
 import orchestrator
 
 
-def handle_exceptions(func):
+def _cli_command(perm):
+    def inner_cli_command(prefix, args="", desc=""):
+        def wrapper(func):
+
+            @wraps(func)
+            def handle_exception(*args, **kwargs):
+                try:
+                    return func(*args, **kwargs)
+                except (orchestrator.OrchestratorError, ImportError) as e:
+                    # Do not print Traceback for expected errors.
+                    return HandleCommandResult(-errno.ENOENT, stderr=str(e))
+                except NotImplementedError:
+                    msg = 'This Orchestrator does not support `{}`'.format(prefix)
+                    return HandleCommandResult(-errno.ENOENT, stderr=msg)
+
+            return CLICommand(prefix, args, desc, perm)(handle_exception)
+        return wrapper
+    return inner_cli_command
 
-    @wraps(func)
-    def inner(*args, **kwargs):
-        try:
-            return func(*args, **kwargs)
-        except (orchestrator.NoOrchestrator, ImportError) as e:
-            return HandleCommandResult(-errno.ENOENT, stderr=str(e))
-    return inner
 
+_read_cli = _cli_command('r')
```

Comment:
maybe <_read_command> and <_write_command> would be more meaningful ,,, but this is just a question of preferences.

Reply:
I will address that in the `orchestrator wait` pr.